File: 20-Valid-Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> brackets;

        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '(' || s[i] == '[' || s[i] == '{')
                brackets.push(s[i]);
            else {
                if (brackets.empty() ||
                    s[i] == ')' && brackets.top() != '(' ||
                    s[i] == '}' && brackets.top() != '{' ||
                    s[i] == ']' && brackets.top() != '[')
                    return false;
                brackets.pop();
            }
        }
        return brackets.empty();
    }
};

File: 20-Valid-Parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let bracketStack = [];
    let bracketPairs = {
        ')': '(',
        '}': '{',
        ']': '['
    };

    for (let char of s) {
        if (char === '(' || char === '{' || char === '[') {
            bracketStack.push(char);
        } else if (char === ')' || char === '}' || char === ']') {
            if (!bracketStack.length || bracketStack.pop() !== bracketPairs[char])
                return false;
        }
    }
    return !bracketStack.length;
};